### PR TITLE
Liveblog widgets

### PR DESCRIPTION
**Pull request by callumlocke**
- [x] state of play
- [x] votes vs seats
- [ ] single constituency lookup
- [ ] exit polls data (for the first few hours)
- [ ] cartogram
- [ ] 'what happens now' infobox/link

### CODE REVIEW

File: server/util/filters.js [hunk 1]
```diff
@@ -112,6 +112,10 @@ exports.ftdate = function(date) {
   });
 };
 
+exports.toOneDP = function (number, places) {
+  return (Math.round(number * 10) / 10) + '';
```
Comment (kavanagh) [inline server/util/filters.js]: * Places is not used - this would be more useful if took a places 
* and style: why do `+ ''` when `toString()` is much clearer?

File: server/routers/data/votes-vs-seats.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
+'use strict';
```
Comment (kavanagh) [inline server/routers/data/votes-vs-seats.js]: could this be moved to server/data/votes-vs-seats.js

File: server/routers/data/state-of-play.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
+'use strict';
```
Comment (kavanagh) [inline server/routers/data/state-of-play.js]: Could this be moved to `server/data/state-of-play.js`

File: server/index.js [hunk 4]
```diff
@@ -34,6 +35,12 @@ function main() {
     this.set('Timing-Allow-Origin', '*');
     yield next;
   });
+
+
+  app.use(cors({
```
Comment (kavanagh) [inline server/index.js]: Why is this needed?

Could this be moved to a route specific middleware?

File: server/index.js [hunk 5]
```diff
@@ -34,6 +35,12 @@ function main() {
...
+    origin: true // do something more specific here
```
Comment (kavanagh) [inline server/index.js]: ` // do something more specific here` - what does this mean? Is it copy-n-paste code?

File: server/routers/data/state-of-play.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+var ukParties = require('uk-political-parties');
+var _ = require('lodash');
+
+var knownParties = [
```
Comment (kavanagh) [inline server/routers/data/state-of-play.js]: Could this be something you add to the uk-political-parties library?

e.g. 
* `ukParties.getKnownParties()` return an array of IDs. and
* `ukParties.isKnownParty(id)` returns a boolean
*  `ukParties.isOther(id or name)` returns boolean

File: client/scss/widgets/_votes-vs-seats.scss [hunk 7]
```diff
@@ -0,0 +1,75 @@
+$vvs-bar-height: 10;
```
Comment (kavanagh) [inline client/scss/widgets/_votes-vs-seats.scss]: we already have a `components` directory for this, so this should be `client/scss/components/_votes-vs-seats.scss`

File: client/scss/widgets/_state-of-play.scss [hunk 8]
```diff
@@ -0,0 +1,101 @@
+.state-of-play {
```
Comment (kavanagh) [inline client/scss/widgets/_state-of-play.scss]: use `client/scss/components/` directory instead.

File: client/scss/_functions.scss [hunk 9]
```diff
@@ -0,0 +1,3 @@
+@function color($name) {
```
Comment (kavanagh) [inline client/scss/_functions.scss]: a microoptimisation?

File: server/index.js (hunk 4 above)
Comment (callumlocke) [inline server/index.js]: Without this, the liveblog page refused to load liveblog.js.

What's the correct way to make it route-specific?

File: server/index.js (hunk 5 above)
Comment (callumlocke) [inline server/index.js]: `origin: true` allows all origins. My comment is a reminder to do something less permissive, eg whitelist training.blogs.ft.com and blogs.ft.com - wasn't immediately obvious from [koa-cors](https://github.com/evert0n/koa-cors) docs how to do this and I didn't want to waste time on it since you'd probably have an opinion on how to best handle this

File: server/routers/data/state-of-play.js (hunk 6 above)
Comment (callumlocke) [inline server/routers/data/state-of-play.js]: Could do, but I'm using this hardcoded array purely to assist with generating my dummy data – probably won't be needed once I switch over to reading real data out of the Loki DB?

File: server/index.js (hunk 4 above)
Comment (kavanagh) [inline server/index.js]: So could you just move this down onto the liveblog router or even the specific route.

Routes can take multiple middleware generators. eg

```javascript
router.get('/liveblog/thingy', cors(), thingy());
```

File: client/scss/_functions.scss (hunk 9 above)
Comment (callumlocke) [inline client/scss/_functions.scss]: It's a shortcut for readability. I've got `color()` all over my code. I can remove it if you want, but personally I find SCSS much harder to skim when it's got `oColorsGetPaletteColor()` all over the place.

File: server/util/filters.js (hunk 1 above)
Comment (callumlocke) [inline server/util/filters.js]: Can swig filters take arguments?

Comment (kavanagh) [inline server/util/filters.js]: Yes.

Comment (callumlocke) [inline server/util/filters.js]: Yes they can.

File: server/index.js (hunk 4 above)
Comment (callumlocke) [inline server/index.js]: But it's also needed for the stuff in public – `/js/liveblog.js`, `/css/pages/liveblog.css`

File: server/util/filters.js (hunk 1 above)
Comment (callumlocke) [inline server/util/filters.js]: :+1: done

File: server/routers/data/votes-vs-seats.js (hunk 2 above)
Comment (callumlocke) [inline server/routers/data/votes-vs-seats.js]: :+1: done

File: server/routers/data/state-of-play.js (hunk 3 above)
Comment (callumlocke) [inline server/routers/data/state-of-play.js]: :+1: done

File: client/scss/widgets/_votes-vs-seats.scss (hunk 7 above)
Comment (callumlocke) [inline client/scss/widgets/_votes-vs-seats.scss]: :+1: done

File: client/scss/widgets/_state-of-play.scss (hunk 8 above)
Comment (callumlocke) [inline client/scss/widgets/_state-of-play.scss]: :+1: done

**Comment by kavanagh** [comment]
@callumlocke could you also make all the widget responsive/fluid so they can be popped into the final results page.
